### PR TITLE
Catalog mapping for Samples and Analyses

### PR DESCRIPTION
**Pull request by ramonski**
## Description of the issue/feature this PR addresses

This PR remaps the sample and analysis catalogs to avoid objects to be indexed in `portal_catalog`.

## Current behavior before PR

Samples and Analyses were indexed in `portal_catalog` as well

## Desired behavior after PR is merged

Samples and Analyses are no longer indexed in `portal_catalog`.

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: src/senaite/core/setuphandlers.py [hunk 1]
```diff
@@ -105,62 +112,65 @@ def getNonInstallableProfiles(self):
...
-    ("SamplePoint", ["senaite_catalog_setup", "portal_catalog"]),
-    ("SampleType", ["senaite_catalog_setup", "portal_catalog"]),
-    ("SamplingDeviation", ["senaite_catalog_setup", "portal_catalog"]),
-    ("StorageLocation", ["senaite_catalog_setup", "portal_catalog"]),
-    ("SubGroup", ["senaite_catalog_setup", "portal_catalog"]),
-    ("Supplier", ["senaite_catalog_setup", "portal_catalog"]),
-    ("WorksheetTemplate", ["senaite_catalog_setup", "portal_catalog"]),
+    ("ARReport", [REPORT_CATALOG, PORTAL_CATALOG]),
+    ("ARTemplate", [SETUP_CATALOG, PORTAL_CATALOG]),
+    ("ARTemplate", [SETUP_CATALOG, PORTAL_CATALOG]),
```
Comment (xispa) [inline src/senaite/core/setuphandlers.py]: This line is duplicate

File: src/senaite/core/setuphandlers.py [hunk 2]
```diff
@@ -105,62 +112,65 @@ def getNonInstallableProfiles(self):
...
+    ("ReferenceSample", [SENAITE_CATALOG, PORTAL_CATALOG]),
+    ("SampleCondition", [SETUP_CATALOG, PORTAL_CATALOG]),
+    ("SampleMatrix", [SETUP_CATALOG, PORTAL_CATALOG]),
+    ("SamplePoint", [SETUP_CATALOG, PORTAL_CATALOG]),
+    ("SampleType", [SETUP_CATALOG, PORTAL_CATALOG]),
+    ("SamplingDeviation", [SETUP_CATALOG, PORTAL_CATALOG]),
+    ("StorageLocation", [SETUP_CATALOG, PORTAL_CATALOG]),
+    ("SubGroup", [SETUP_CATALOG, PORTAL_CATALOG]),
+    ("Supplier", [SETUP_CATALOG, PORTAL_CATALOG]),
+    ("WorksheetTemplate", [SETUP_CATALOG, PORTAL_CATALOG]),
```
Comment (xispa) [inline src/senaite/core/setuphandlers.py:192]: Some content types are missing, like `Worksheet`, `DuplicateAnalysis`, `ReferenceAnalysis`, `Contact`, `InstrumentCalibration`, etc.

**Review by xispa — CHANGES_REQUESTED** [review]
There are some missing content types

File: src/senaite/core/setuphandlers.py (hunk 2 above)
Comment (ramonski) [reply src/senaite/core/setuphandlers.py:192]: The other types were added by the setuphandler `setup_core_catalogs` which maps the types from here: https://github.com/senaite/senaite.core/blob/9adcd931ecec4c9f8cef26cfd0117dda71dc4dff/src/senaite/core/catalog/analysis_catalog.py#L73-L78

However, for completeness I added the missing types in  9adcd931ecec4c9f8cef26cfd0117dda71dc4dff from here:
https://github.com/senaite/senaite.core/blob/2.x/src/bika/lims/__init__.py#L57-L121

File: src/senaite/core/setuphandlers.py (hunk 1 above)
Comment (ramonski) [reply src/senaite/core/setuphandlers.py]: done

**Review by xispa — APPROVED** [review]

**Pull request merged**